File: Data Structures & Algorithms/car-fleet/submission-2.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        # Evaluate the list of time cars will take to reach
        N = len(position)
        time = [0]*N
        for i in range(N):
            time[i] = (target - position[i])/speed[i]
        # Sort the cars and time in terms of position
        sorted_time = [val for _, val in sorted(zip(position, time))]
        print(sorted_time)
        num_fleet = 1
        for i in range(N-1, 0, -1):
            if sorted_time[i] < sorted_time[i-1]:
                num_fleet += 1
            else:
                sorted_time[i-1] = sorted_time[i]
        return num_fleet

File: Data Structures & Algorithms/validate-parentheses/submission-2.cpp
class Solution {
public:
    bool isValid(string s) {
        if (s.size()%2 != 0)
        {
            return false;
        }
        int A[s.size()] = {};
        for (int i = 0; i < s.size(); ++i)
        {
            if (s[i] == '(')
            {
                A[i] = 1;
            }
            else if (s[i] == ')')
            {
                A[i] = -1;
            }
            else if (s[i] == '{')
            {
                A[i] = 2;
            }
            else if (s[i] == '}')
            {
                A[i] = -2;
            }
            else if (s[i] == '[')
            {
                A[i] = 3;
            }
            else if (s[i] == ']')
            {
                A[i] = -3;
            }
        }
        int Char[s.size()/2];
        int m = 0;
        for (int i = 0; i < s.size(); ++i)
        {
            if (A[i] > 0)
            {
                Char[m] = A[i];
                ++m;
            }
            else
            {
                --m;
                if (A[i] != -Char[m])
                {
                    return false;
                }
            }
        }
        if (m == 0)
            return true;
        else
            return false;

    }
};
